examples/inference-deployments/mpt/mpt_7b_ft_handler.py

```diff
@@ -124,30 +131,50 @@ def __init__(self,
             use_attention_linear_bias=use_attention_linear_bias,
             has_positional_encoding=has_positional_encoding,
             shared_contexts_ratio=shared_contexts_ratio)
+        print(f'Loading FT checkpoint from {model_path}')
         if not self.model.load(ckpt_path=model_path):
             raise RuntimeError(
                 'Could not load model from a FasterTransformer checkpoint')
 
         self.tokenizer = AutoTokenizer.from_pretrained(self.model_name,
                                                        trust_remote_code=True)
+        print('FT initialization complete')
 
-    def _parse_inputs(self, inputs: Dict[str, Any]):
+    def _parse_single_input(
+            self, inputs: Dict[str, Any]) -> Tuple[List[str], Dict[str, Any]]:
+        """Splits request into input strings and kwargs."""
         if self.INPUT_STRINGS_KEY not in inputs or not isinstance(
                 inputs[self.INPUT_STRINGS_KEY], list):
             raise RuntimeError(
                 'Input strings must be provided as a list to generate call')
 
-        generate_input = inputs[self.INPUT_STRINGS_KEY]
-        batch_size = len(generate_input)
+        generate_inputs = inputs[self.INPUT_STRINGS_KEY]
 
-        # Set default generate kwargs
+        # Start with defaults and overwrite with request args. We need to use
+        # defaults here so we can detect disagreeing args when we merge across
+        # requests in the batch.
         generate_kwargs = copy.deepcopy(self.DEFAULT_GENERATE_KWARGS)
-
-        # If request contains any additional kwargs, add them to generate_kwargs
         for k, v in inputs.items():
             if k not in [self.INPUT_STRINGS_KEY]:
                 generate_kwargs[k] = v
-        generate_kwargs['top_k'] *= torch.ones(batch_size, dtype=torch.int32)
+
+        return generate_inputs, generate_kwargs
+
+    def _convert_kwargs(self, generate_inputs: List[str],
+                        generate_kwargs: Dict[str, Any]):
+        """Converts generate_kwargs into required torch types."""
+        batch_size = len(generate_inputs)
+
+        # Allow 'max_length' to be an alias for 'output_len'. Makes it less
+        # likely clients break when we swap in the FT handler.
+        if 'max_length' in generate_kwargs:
+            generate_kwargs['output_len'] = generate_kwargs['max_length']
```

Comment:
wrap this by int() too.

Reply:
that's redundant, right? since we'll always convert `output_len` on line 175.

Reply:
True. Missed that.